x/ccv/child/keeper/distribution.go

```diff
@@ -0,0 +1,69 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/modules/core/04-channel/types"
+	"github.com/cosmos/interchain-security/x/ccv/child/types"
+	// XXX delete or uncomment this block before merge
+	//ibctypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+)
+
+// Simple model, donate tokens to the fee pool of the provider validator set
+// reference: cosmos/ibc-go/modules/apps/transfer/keeper/msg_server.go
+func (k Keeper) DonateToProviderValidatorSet(ctx sdk.Context) error {
+	if !k.shouldTransmit(ctx) {
+		return nil
+	}
+
+	port := k.GetDistributionTransmissionPort(ctx)
+	channel := k.GetDistributionTransmissionChannel(ctx)
+
+	// work around to reuse the IBC token transfer logic
+	consumerFeePoolAddr := k.accountKeeper.GetModuleAccount(ctx, k.feeCollectorName).GetAddress()
+	providerFeePoolAddrStr := k.GetProviderFeePoolAddrStr(ctx)
+
+	tokens := k.bankKeeper.GetAllBalances(ctx, consumerFeePoolAddr)
+	for _, token := range tokens {
+
+		// XXX delete or uncomment this block before merge
+		//if !strings.HasPrefix(token.Denom, "ibc/") {
+		//    denomTrace := ibctypes.ParseDenomTrace(token.Denom)
+		//    token.Denom = denomTrace.IBCDenom()
+		//}
+
+		err := k.ibcTransferKeeper.SendTransfer(ctx,
+			port,
+			channel,
+			token,
+			consumerFeePoolAddr,
+			providerFeePoolAddrStr,
+			clienttypes.Height{0, 0}, // XXX timeoutHeight
+			0,                        // XXX timeoutTimestamp
+		)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// test to see if enough blocks have passed for
+// a transmission to occur
+func (k Keeper) shouldTransmit(ctx sdk.Context) bool {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.LastDistributionTransmissionKey)
+	if bz == nil {
+		return true
+	}
+	ltbh := &types.LastTransmissionBlockHeight{}
+	ltbh.Unmarshal(bz)
+	bpdt := k.GetBlocksPerDistributionTransmission(ctx)
+	curHeight := ctx.BlockHeight()
+	return (curHeight - ltbh.Height) >= bpdt
+}
+
+func (k Keeper) ChannelOpenInit(ctx sdk.Context, msg *channeltypes.MsgChannelOpenInit) (
+	*channeltypes.MsgChannelOpenInitResponse, error) {
+	return k.ibcCoreKeeper.ChannelOpenInit(ctx, msg)
```

Comment:
@AdityaSripal I'm not sure if we're allowed to use `context.Context` and `sdk.Context` interchangeably here or if this is just a dep issue? I get a compilation error here: 
```
x/ccv/child/keeper/distribution.go:68:40: cannot use ctx (type "github.com/cosmos/cosmos-sdk/types".Context) as type context.Context in argument to k.ibcCoreKeeper.ChannelOpenInit:
	"github.com/cosmos/cosmos-sdk/types".Context does not implement context.Context (missing Deadline method)
```
Thoughts on resolving this error?

Reply:
They are different types.

You need to wrap sdk.Context: https://github.com/cosmos/cosmos-sdk/blob/a0a1197c2786647139bab92fcc6f9f3ee0d867d5/types/context.go#L278

Reply:
Oh yeah sweet. Thanks